src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,132 @@
 package core.basesyntax;
 
-public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+import java.util.NoSuchElementException;
+
+public class ArrayList<T> implements MyList<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double MULTIPLIER = 1.5;
+    private static final String MESSAGE_OUT_OF_BOUNDS = "The index passed to any of "
+            + "the methods is invalid";
+    private static final String MESSAGE_NO_ELEMENT = "There is no such element";
+
+    private T[] elements;
+    private int size = 0;
 
+    @SuppressWarnings("unchecked")
+    public ArrayList() {
+        elements = (T[]) new Object[DEFAULT_CAPACITY];
     }
 
     @Override
-    public void add(T value, int index) {
-
+    public boolean add(T value) {
+        increaseCapacityIfFull();
+        elements[size++] = (T) value;
```

Comment:
The cast to (T) in 'elements[size++] = (T) value;' is unnecessary since 'value' is already of type T. This is a minor issue but can be cleaned up for clarity.